test/src/e2e.dynval/dv.n+1.test.ts

```diff
@@ -186,6 +186,51 @@ describe("Dynamic Validator N -> N+1", function() {
         });
     });
 
+    describe("Delegate more stake to which was less than the minimum delegation", async function() {
+        const nodes = withNodes(this, {
+            promiseExpect,
+            validators: otherDynValidators
+                .map((signer, index) => ({
+                    signer,
+                    delegation: 5_000,
+                    deposit: 10_000_000 - index // tie-breaker
+                }))
+                .concat([
+                    {
+                        signer: betty,
+                        delegation: 999,
+                        deposit: 10_000_000
+                    }
+                ]),
+        });
+
+        it("betty should be included in validators", async function() {
+            const checkingNode = nodes[0];
+            await beforeInsertionCheck(checkingNode.sdk);
```

Comment:
It's being called in every test suites in this file. How about moving this to `beforeEach`?

Reply:
nodes are now being initialized in every child `describe`. To guarantee the `beforeInsertionChecking` is invoked after the nodes initialization, I should add `beforeEach` hooks for every child `describe`. Currently, I have only one `it` in every child `describe`, I thought I don't have to use `beforeEach`.
Inserting one top level `beforeEach` is not possible.

Reply:
What a pity. :(